### PR TITLE
Add GitHub action for pre-commit

### PR DESCRIPTION
**Pull request by dhruvmanila**
Follow-up on #2511 
As mentioned in https://github.com/TheAlgorithms/Python/pull/2511#issuecomment-701163537

### **Describe your change:**

* [x] Add an algorithm?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.

### CODE REVIEW

**Comment by dhruvmanila** [comment]
It's working perfectly: https://github.com/TheAlgorithms/Python/pull/2515/checks?check_run_id=1186742542

A few small changes required:
- Fix `instagram_crawler.py` using `black`
- Either add `shebang` or remove the executable bit from those files

**Comment by dhruvmanila** [comment]
I think the name of the file should be ***lint***, what do you think? @cclauss

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+
```
Comment (cclauss) [inline .github/workflows/pre-commit.yml:11]:
```suggestion
```

Comment (cclauss) [reply .github/workflows/pre-commit.yml:11]: Not needed.

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
```
Comment (cclauss) [inline .github/workflows/pre-commit.yml:19]:
```suggestion
      - uses: actions/setup-python@v2
```

File: .github/workflows/pre-commit.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Install dependencies
```
Comment (cclauss) [inline .github/workflows/pre-commit.yml:20]:
```suggestion
      - name: Install pre-commit
```

File: .github/workflows/pre-commit.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade pre-commit
+
+      - name: Run pre-commit
+        run: pre-commit run --verbose --all-files --show-diff-on-failure
```
Comment (cclauss) [inline .github/workflows/pre-commit.yml:26]:
```suggestion
      - run: pre-commit run --verbose --all-files --show-diff-on-failure
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Simplifying by always using the latest and greatest Python and by dropping names that are less self-documenting than the commands that they run.

**Comment by cclauss** [comment]
Why not just add the shebang instead of removing the execution bit?

**Comment by cclauss** [comment]
> I think the name of the file should be _**lint**_, what do you think? @cclauss

I think the current name is good enough.  The script runs pre-commit so it should be called pre-commit.

**Comment by dhruvmanila** [comment]
Except for the `scheduling/round_robin.py` file, other files don't output anything when I run `python <filename>` so I thought there's no point in keeping the executable bit on. 

If you think otherwise, I will turn them on?

**Review by cclauss — APPROVED** [review]

**Pull request merged**